src/rulesets/Base/array.jl

```diff
@@ -65,9 +65,32 @@ function rrule(::typeof(reshape), A::AbstractArray, dims::Union{Colon,Int}...)
     return reshape(A, dims...), reshape_pullback
 end
 
+#####
+##### `permutedims`
+#####
+
+function rrule(::typeof(permutedims), x::AbstractVector)
+    project = ProjectTo(x)
+    permutedims_pullback_1(dy) = (NoTangent(), project(permutedims(unthunk(dy))))
+    return permutedims(x), permutedims_pullback_1
+end
+
+function rrule(::typeof(permutedims), x::AbstractArray, perm)
+    pr = ProjectTo(x)  # projection restores e.g. transpose([1,2,3])
+    permutedims_back_2(dy) = (NoTangent(), pr(permutedims(unthunk(dy), invperm(perm))), NoTangent())
+    return permutedims(x, perm), permutedims_back_2
+end
+
+function rrule(::typeof(PermutedDimsArray), x::AbstractArray, perm)
+    pr = ProjectTo(x)
+    permutedims_back_3(dy) = (NoTangent(), pr(permutedims(unthunk(dy), invperm(perm))), NoTangent())
+    return PermutedDimsArray(x, perm), permutedims_back_3
+end
```

Comment:
BTW, the logic of doing `permutedims` in reverse pass is this. Calling PermutedDimsArrray forwards means you think the next operation will be OK, something like broadcasting, not like reshape + BLAS. But the previous operation might not be, we don't know that it's gradient is going to like a lazy container. So I think it's safest to copy.

These could all have Inplaceabelthunks, with `permutedims!`, but perhaps we kick that can down the road until they are useful.